crates/core/src/transport/peer_connection.rs

```diff
@@ -137,7 +147,66 @@ impl PeerConnection {
             failure_count: 0,
             first_failure_time: None,
             last_packet_report_time: Instant::now(),
-        }
+            keep_alive_handle: None,
+        };
+
+        // Start the keep-alive task immediately
+        conn.start_keep_alive();
```

Comment:
can't we just start the background task and instance Self with the handle instead of having to mutate self after instancing it just to add the handle?